src/main/java/com/mparticle/kits/IterableDeviceIdHelper.java

```diff
@@ -0,0 +1,35 @@
+package com.mparticle.kits;
+
+import android.content.Context;
+import android.provider.Settings;
+
+import java.lang.reflect.Method;
+
+class IterableDeviceIdHelper {
+    static String getGoogleAdId(Context context) {
```

Comment:
this API shouldn't be called on main thread - Android/Google Play has started throwing errors for this. Also - reflection the main thread should be avoided.

Reply:
Made this asynchronous.